src/lexer.hpp

```diff
@@ -52,4 +48,15 @@ struct Lexer {
 	Token const& peek_token(int dt = 0) {
 		return token_at(m_token_index + dt);
 	}
+
+	std::vector<char> m_source;
+	TokenArray& m_tokens;
+
+	Trie m_symbols_trie;
+
+	int m_source_index {0};
+	int m_token_index {0};
+
+	int m_current_line {0};
+	int m_current_column {0};
```

Comment:
Why did you move this down here?

Reply:
I draw attention to this because it is inconsistent with the rest of the codebase.

Reply:
I think we should move to that style everywhere.

When you look at a class definition, you more often want to know what the methods are. It's just more convenient to read.